openthread-2018-02-27/build.sh

```diff
@@ -1,12 +1,17 @@
 #!/bin/bash
 # Copyright 2018 Google Inc. All Rights Reserved.
 # Licensed under the Apache License, Version 2.0 (the "License");
+. $(dirname $0)/../custom-build.sh $1 $2
 . $(dirname $0)/../common.sh
 
 build_lib() {
   rm -rf BUILD
   cp -rf SRC BUILD
   [[ -f $LIB_FUZZING_ENGINE ]] && cp $LIB_FUZZING_ENGINE BUILD/tests/fuzz/
+  if [[ $FUZZING_ENGINE == "hooks" ]]; then
+    # Link ASan runtime so we can hook memcmp et al.
+    LIB_FUZZING_ENGINE="$LIB_FUZZING_ENGINE -fsanitize=address"
+  fi
   (cd BUILD && ./bootstrap && ./configure \
     --enable-fuzz-targets               \
     --enable-application-coap           \
```

Comment:
I guess this lib make files know about LIB_FUZZING_ENGINE?

Reply:
Yes, openthread links with `LIB_FUZZING_ENGINE` after building the fuzz target.  Adding `-fsanitize=address` here might be a hack, but I couldn't think of a better alternative.  We don't want to build everything with ASan, just link with it at the end to get its `memcmp`, `strcmp`, etc. interceptors.